lib/promscrape/targetstatus.go

```diff
@@ -15,6 +15,11 @@ func WriteHumanReadableTargetsStatus(w io.Writer) {
 	tsmGlobal.WriteHumanReadable(w)
 }
 
+//return the number of target
+func GetTargetNumber() int {
+	return len(tsmGlobal.m)
```

Comment:
This is racy access to `tsmGlobal.m`, since another goroutine can update the map during this access. Such an access must be protected by `tsmGlobal.mu`.